src/flamegraph/color.rs

```diff
@@ -0,0 +1,272 @@
+use std::str::FromStr;
+use std::collections::HashMap;
+use std::fs::OpenOptions;
+use std::io;
+use std::io::Write;
+use std::path::Path;
+use std::fs::File;
+use std::io::BufReader;
+use std::io::BufRead;
+
+pub(super) const VDGREY: &str = "rgb(160,160,160)";
+pub(super) const DGREY: &str = "rgb(200,200,200)";
+
+const YELLOW_GRADIENT: (&str, &str) = ("#eeeeee", "#eeeeb0");
+const BLUE_GRADIENT: (&str, &str) = ("#eeeeee", "#e0e0ff");
+const GRAY_GRADIENT: (&str, &str) = ("#f8f8f8", "#e8e8e8");
+
+#[derive(Debug, PartialEq)]
+pub enum Palette {
+    Hot,
+    Mem,
+    Io,
+    Wakeup,
+    Chain,
+    Java,
+    Js,
+    Perl,
+    Red,
+    Green,
+    Blue,
+    Aqua,
+    Yellow,
+    Purple,
+    Orange,
+}
+
+impl Default for Palette {
+    fn default() -> Self {
+        Palette::Hot
+    }
+}
+
+impl FromStr for Palette {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hot" => Ok(Palette::Hot),
+            "mem" => Ok(Palette::Mem),
+            "io" => Ok(Palette::Io),
+            "wakeup" => Ok(Palette::Wakeup),
+            "chain" => Ok(Palette::Chain),
+            "java" => Ok(Palette::Java),
+            "js" => Ok(Palette::Js),
+            "perl" => Ok(Palette::Perl),
+            "red" => Ok(Palette::Red),
+            "green" => Ok(Palette::Green),
+            "blue" => Ok(Palette::Blue),
+            "aqua" => Ok(Palette::Aqua),
+            "yellow" => Ok(Palette::Yellow),
+            "purple" => Ok(Palette::Purple),
+            "orange" => Ok(Palette::Orange),
+            unknown=> Err(format!("unknown color palette: {}", unknown))
+        }
+    }
+}
+
+fn namehash(name: &str) -> f32 {
+    let mut vector = 0f32;
+    let mut weight = 1f32;
+    let mut max = 1f32;
+    let mut modulo = 10;
+
+    let name = {
+        let looking_for_module_name = if name.starts_with("`") {
+            &name[1..]
+        } else {
+            name
+        };
+
+        if let Some(index) = looking_for_module_name.find("`") {
+            &looking_for_module_name[index + 1..]
+        } else {
+            name
+        }
+    };
+
+    for character in name.bytes() {
+        let i = (character % modulo) as f32;
+        vector += (i / ((modulo - 1) as f32)) * weight;
+        modulo += 1;
+        max += weight;
+        weight *= 0.70;
+
+        if modulo > 12 {
+            break
+        }
+    }
+
+    (1f32 - vector / max)
+}
+
+fn handle_java_palette(s: &str) -> Palette {
```

Comment:
I think I'd prefer to see these functions each in their own sub-module as a `resolve` function:
```rust
colors::java::resolve(name: &str) -> Palette
```

Reply:
Do you prefer to add a subdirectory `colors`, and files `java.rs`, `js.rs`, `perl.rs`, etc... or submodules in the `color.rs` file ?
I find the subdirectory approach a little overkill for a single, relatively small, function per submodule.

Reply:
I think what I'd do is make a file `palettes.rs`, and in `color.rs` do:
```rust
mod palettes;
```
And then in `palettes.rs` do:
```rust
mod java {
  fn resolve(name: &str) -> Palette {}
}
mod perl {
  fn resolve(name: &str) -> Palette {}
}
```
etc. That strikes a good middle-ground.

Reply:
The `palettes` mod will be defined in `flamegraph/mod.rs`, not `color.rs` right ?

Reply:
[Implemented here](https://github.com/jonhoo/inferno/pull/32/commits/ad146b62196810a94e99b97231c57fc463293b59)